pkgs/stdenv/generic/setup.sh

```diff
@@ -660,15 +660,11 @@ export NIX_INDENT_MAKE=1
 # means that we're supposed to try and auto-detect the number of
 # available CPU cores at run-time.
 
-if [ -z "${NIX_BUILD_CORES:-}" ]; then
-  NIX_BUILD_CORES="1"
-elif (( NIX_BUILD_CORES <= 0 )); then
-  NIX_BUILD_CORES=$(nproc 2>/dev/null || true)
-  if expr >/dev/null 2>&1 "$NIX_BUILD_CORES" : "^[0-9][0-9]*$"; then
-    :
-  else
-    NIX_BUILD_CORES="1"
-  fi
+NIX_BUILD_CORES="${NIX_BUILD_CORES:-1}"
+if ((NIX_BUILD_CORES <= 0 )); then
+  GUESS=$(nproc 2>/dev/null || true)
+  NUMERIC_GUESS="${GUESS//[^0-9]/}"
+  NIX_BUILD_CORES="${NUMERIC_GUESS:-1}"
```

Comment:
Slightly simpler (though I’d have used a lowercase name for `GUESS`, or reused `NIX_BUILD_CORES` as before):

```suggestion
  ((NIX_BUILD_CORES = GUESS <= 0 ? 1 : GUESS))
```

Reply:
I like it! Let me test quickly.